src/java/com/zimbra/oauth/utilities/LdapConfiguration.java

```diff
@@ -0,0 +1,143 @@
+                        if (s.length == 3 && s[2].equals(appName)) {
+                            value = s[0];
+                            break;
+                        }
+                    }
+                }
+            } else if (key.endsWith(OAuth2Constants.OAUTH_CLIENT_SECRET)) {
+                String[] registeredOAuth2Clients = Provisioning.getInstance().getConfig()
+                    .getMultiAttr(Provisioning.A_zimbraOAuthConsumerCredentials);
+                if (registeredOAuth2Clients != null && registeredOAuth2Clients.length != 0) {
+                    // {consumer-id}:{secret}:{consumer-app-name}
+                    for (String consumer : registeredOAuth2Clients) {
+                        String s[] = consumer.split(":");
+                        if (s.length == 3 && s[2].equals(appName)) {
+                            value = s[1];
+                            break;
+                        }
+                    }
+                }
+            } else if (key.endsWith(OAuth2Constants.OAUTH_CLIENT_REDIRECT_URI)) {
+                String[] registeredOAuth2RedirectUrls = Provisioning.getInstance().getConfig()
+                    .getMultiAttr(Provisioning.A_zimbraOAuthConsumerRedirectUri);
+                if (registeredOAuth2RedirectUrls != null
+                    && registeredOAuth2RedirectUrls.length != 0) {
+                    // {redirectURI}:{consumer-app-name}
+                    for (String consumer : registeredOAuth2RedirectUrls) {
+                        String s[] = consumer.split(":");
+                        if (s.length == 2 && s[1].equals(appName)) {
+                            return s[0];
+                        }
+                    }
+                }
+            } else if (key.endsWith(OAuth2Constants.OAUTH_SCOPE)) {
+                String[] registeredOAuth2APIScope = Provisioning.getInstance().getConfig()
+                    .getMultiAttr(Provisioning.A_zimbraOAuthConsumerAPIScope);
+
+                if (registeredOAuth2APIScope != null && registeredOAuth2APIScope.length != 0) {
+                    for (String consumer : registeredOAuth2APIScope) {
+                        String s[] = consumer.split(":");
+                        if (s.length == 2 && s[1].equals(appName)) {
+                            value = s[0];
+                            break;
+                        }
+                    }
+                }
+            }
+        } catch (ServiceException e) {
+            ZimbraLog.extensions.info("Error fetching configuration : %s for : %s", key, appName);
+            ZimbraLog.extensions.debug(e);
+        }
+        return value;
+        
+    }
+
+
+    /**
+     * Return the name of the client app, based on the key.
+     * @param key the client related key(zm_oauth_yahoo_client_id,zm_oauth_google_client_secret,
+     * zm_oauth_outlook_client_redirect_uri
+     * @return the client app name
+     */
+    private String getAppName(String key) {
```

Comment:
The property `clientId` from the parent class already knows which social service this instance of the (ldap)configuration class is for, so you don't need this method. Just make clientId `protected` and use it directly, or call the public method `this.getClientId`.

Reply:
Fixed